packages/phpstan-rules/packages/cognitive-complexity/src/Rules/ClassLikeCognitiveComplexityRule.php

```diff
@@ -38,9 +39,11 @@ final class ClassLikeCognitiveComplexityRule extends AbstractSymplifyRule implem
      */
     public function __construct(
         private AstCognitiveComplexityAnalyzer $astCognitiveComplexityAnalyzer,
+        private CompositionOverInheritanceAnalyzer $compositionOverInheritanceAnalyzer,
         private SimpleNameResolver $simpleNameResolver,
         private int $maxClassCognitiveComplexity = 50,
-        array $limitsByTypes = []
+        array $limitsByTypes = [],
+        private bool $scoreCompositionOverInheritance = false,
```

Comment:
with this new paramter one can activate the composition-over-inheritance based analysis.

I left it OFF-by-default for BC